galsim/config/util.py

```diff
@@ -379,19 +379,21 @@ def ParseRandomSeed(config, param_name, base, seed_offset):
     """
     from .value import ParseValue
 
-    # Normally, random_seed parameter is just a number, which really means to use that number
-    # for the first item and go up sequentially from there for each object.
+    # Normally, random_seed parameter is just a number, which really means to use that seed
+    # start a sequential sequence of seeds for each object.
     # However, we allow for random_seed to be a gettable parameter, so for the
     # normal case, we just convert it into a Sequence.
-    if isinstance(config[param_name], int):
-         # The "first" is actually the seed value to use for anything at file or image scope
-         # using the obj_num of the first object in the file or image.  Seeds for objects
-         # will start at 1 more than this.
+    if isinstance(config[param_name], (int, str)):
```

Comment:
IDK why strings get done here. I think this is related to the gotcha above?

Reply:
I think the issue is that the code conflates two things:

1. the seed being an integer type 
2. the sequence of seeds is setup properly as a sequence type

So my guess is that we simply need to make the if statement more precise with something like:

```
if not (isinstance(config[param_name], dict) and "type" in config[param_name] and config[param_name]["type"] == "Sequence"):
```

Reply:
That might even fail for custom seed specifications where part of the sequence definition also needs to be parsed. We could set a special `__rng_is_setup` variable or something.

Reply:
It definitely doesn't have to be explicitly a Sequence.  E.g. your guard would not work right for demo13.  (Either the old or new version).

Reply:
FWIW, I think the only potential gotcha left in the version I ended up with in this PR is if someone wanted to make an rng sequence using an Eval and wanted to do it using the shorthand `$` notation (as we used to do in demo13.  That's now doubly documented -- both in the demo script where it would have happened and in the Sphinx docs.  So there's pretty fair warning about it.  
The workaround to avoid it is also pretty simple.  Just don't use the `$` shorthand.

Reply:
Currently:
* (int, str) indicate that this is just a single value, which gets converted into a Sequence.
* (dict) indicates that either the above already happened or the user set it up themselves to do whatever they want.

Reply:
Got it. 

I made a PR into this one with another design: https://github.com/GalSim-developers/GalSim/pull/1173

What I was going for was for the code to attempt to parse the seed as an int. If that passes, then use that int to set up the seed sequence. Otherwise, don't do anything. IDK if this might have other problems though.

Reply:
It only does this once using a variable stored in the config to control the state.

Reply:
I think the current design excludes a case where someone uses a short-hand eval string to indicate a value they'd actually like to be a constant.

Reply:
No.  The opposite.  Right now, a short-hand eval string (i.e. with `$`) is taken to be a constant, which is parsed as a single int and then made into a Sequence.
The current design excludes someone who wants to use some complicated dict to calculate a single int once, and have that value be treated as a constant and turned into a Sequence.  
That seems unlikely, but if someone really wants to do this, they can, by doing this in an eval_variable rather than directly in the image.random_seed field.  Call this `my_first_seed` or whatever.  Then they can use `image.random_seed = $my_first_seed` to evaluate it that one time and then sequence from there.